### PR TITLE
SAC mujoco result

### LINKED ISSUE #209: mujoco examples' result mismatch
With all of ddpg, sac and td3, I get similar results like this:
image

But the official results like this:
image

### LINKED ISSUE #233: Deterministic output for SAC policy
When I ran the example code, I found that even in the testing session, the SAC strategy is still a random policy. Shouldn't the mean value be output as a deterministic policy during the training?

### PR DESCRIPTION
**Pull request by Trinkle23897**

### CODE REVIEW

File: tianshou/policy/modelfree/sac.py [hunk 1]
```diff
@@ -125,6 +130,9 @@ def forward(  # type: ignore
         if self._noise is not None and not self.updating:
             act += to_torch_as(self._noise(act.shape), act)
         act = act.clamp(self._range[0], self._range[1])
+        if self._deterministic_eval and not self.training:
+            act = torch.tanh(logits[0]) * \
+                self._action_scale + self._action_bias
```
Comment (danagi) [inline tianshou/policy/modelfree/sac.py]: Stochastic action is sampled before these lines causing the `log_prob` to be wrong and `act.clamp` to be ignored. Suggest coding like `x = logits[0] if self._deterministic_eval and not self.training else dist.rsample()` and updating the logic of adding noise instead.

**Review by danagi — CHANGES_REQUESTED** [review]

File: tianshou/policy/modelfree/sac.py [hunk 2]
```diff
@@ -130,9 +133,6 @@ def forward(  # type: ignore
         if self._noise is not None and not self.updating:
             act += to_torch_as(self._noise(act.shape), act)
```
Comment (danagi) [inline tianshou/policy/modelfree/sac.py]: action noise is supposed to be added at collecting state (`self.training`=`True` and `self.updating`=`False`) only but it is added at  testing state (`self.training`=`False` and `self.updating`=`False`) also now.

**Review by danagi — CHANGES_REQUESTED** [review]

**Review by danagi — APPROVED** [review]
